### PR TITLE
Back-populate script to add hashed e-mails to Pegasus forms table

### PR DESCRIPTION
**Pull request by ewjordan**
This is step 3 in the comment at https://github.com/code-dot-org/code-dot-org/pull/11400#issuecomment-256695027, and can be merged at any point but cannot be run until https://github.com/code-dot-org/code-dot-org/pull/11424 is live on production and the hashed e-mail field is created.

This has been tested locally, but I added a --dry-run option so that I can verify it would be doing the right thing before actually writing any data.

### CODE REVIEW

File: bin/oneoff/back_populate_forms_hashed_email [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  DB[:forms].
+    where(id: start_index..(start_index + batch_size)).
+    each do |form|
+    hashed_email = Digest::MD5.hexdigest(form[:email])
+    hashed_email_updates[form[:id]] = hashed_email
+  end
+
+  if dry_run
+    hashed_email_updates.each_pair do |id, hashed_email|
+      puts "Would have written #{id} => #{hashed_email}"
```
Comment (ashercodeorg) [inline bin/oneoff/back_populate_forms_hashed_email:41]: This'll output something like O(M) rows (in the production environment)... Do you mind spamming console output that much?

File: bin/oneoff/back_populate_forms_hashed_email [hunk 2]
```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env ruby
+
+# This script populates the hashed_email column of the pegasus forms table.
+# This is based off of back_populate_contacts_hashed_email
+
+require File.expand_path('../../../pegasus/src/env', __FILE__)
```
Comment (ashercodeorg) [inline bin/oneoff/back_populate_forms_hashed_email:6]: Is this necessary? Not that it matters much.

File: bin/oneoff/back_populate_forms_hashed_email [hunk 3]
```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env ruby
```
Comment (ashercodeorg) [inline bin/oneoff/back_populate_forms_hashed_email:1]: Apologies that I didn't think of this before... Do we want to propagate the `hashed_email` column to SOLR? My assumption is that we do not, though I do at least want to raise the question.

(CC @jeremydstone)

File: bin/oneoff/back_populate_forms_hashed_email [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+if ARGV.length < 3 || ARGV.length > 4
+  puts 'Usage: ./bin/oneoff/back_populate_forms_hashed_email '\
+    'start_index end_index batch_size [--dry-run]'
+  exit 1
+end
+
+start_index = ARGV[0].to_i
+end_index = ARGV[1].to_i
+batch_size = ARGV[2].to_i
+dry_run = (ARGV[3] == '--dry-run')
```
Comment (ashercodeorg) [inline bin/oneoff/back_populate_forms_hashed_email]: The parentheses are unnecessary.

**Comment by ashercodeorg** [comment]
LGTM (outside of spamming console). FWIW, I think my `batch_size` was about 5K or 10K when actually run in production.

File: bin/oneoff/back_populate_forms_hashed_email (hunk 2 above)
Comment (ewjordan) [reply bin/oneoff/back_populate_forms_hashed_email:6]: I can see - it was in other files that did similar things, so I didn't check yet.

Comment (ashercodeorg) [reply bin/oneoff/back_populate_forms_hashed_email:6]: Copying my old work will only get you into trouble.  :)

File: bin/oneoff/back_populate_forms_hashed_email (hunk 1 above)
Comment (ewjordan) [reply bin/oneoff/back_populate_forms_hashed_email:41]: I feel like if --dry-run is set, console spam is the expectation, right? That's how it works with most AWS utils. My feeling is that I'd only ever set --dry-run for a very small batch, just to make sure it's doing what I think it should be doing.

Comment (ashercodeorg) [reply bin/oneoff/back_populate_forms_hashed_email:41]: SGTM. Keep as is.

File: bin/oneoff/back_populate_forms_hashed_email (hunk 2 above)
Comment (ewjordan) [reply bin/oneoff/back_populate_forms_hashed_email:6]: Yep, turns out this is necessary, it pulls in some stuff that this won't run without

**Pull request merged**